cmd/signature-v4-utils.go

```diff
@@ -56,7 +56,25 @@ func skipContentSha256Cksum(r *http.Request) bool {
 
 	// If x-amz-content-sha256 is set and the value is not
 	// 'UNSIGNED-PAYLOAD' we should validate the content sha256.
-	return !(ok && v[0] != unsignedPayload)
+	if ok {
+		switch v[0] {
+		case unsignedPayload:
+			return true
+		case emptySHA256:
+			// some broken clients set empty-sha256
+			// with > 0 content-length in the body,
+			// we should skip such clients and allow
+			// blindly such insecure clients.
+			if r.ContentLength > 0 && !globalAPIStrictSha256 {
```

Comment:
Why doesn't this allow `UNSIGNED-PAYLOAD` when content length is 0?

Seems like these clients would also send it on 0 size objects?

Reply:
@klauspost so I found out that instead of `UNSIGNED-PAYLOAD` they set `emptSha256` when contentLength > 0 - if contentLength == 0 we don't really care anyways. 

These kinds of applications are very rare and almost always incorrectly implemented.

Reply:
actually I need to update the comment.. we do not blindly ignore it, we in-fact allow this situation i.e

`x-amz-content-sha256 == emptySha256` while the `r.ContentLength > 0` when strictSha256 is disabled.

Reply:
Also there is no plan to merge this, currently need to test this with the customer with their application.